### DISCUSSION TITLE
How to output/assign Young's modulus and Possion's ration to every quadrature points in the output CSV file

### DISCUSSION THREAD

**Discussion by Yulongggggg** [idaholab/moose#29108] [Q&A Modules: Solid mechanics]
How to output/assign Young's modulus and Poisson's ratio to every quadrature point in the output CSV file, is there a method to implement this? For example, in this image, add two columns named Young's modulus and Poisson's ratio
![image](https://github.com/user-attachments/assets/bedea645-a61a-441a-9ebe-0db03925c483)

**Comment by Yulongggggg** [comment]
@GiudGiud

**Reply by GiudGiud** [reply]
MaterialValueSampler is the way to do this for output

For assignment, you can use a PiecewiseConstantFromCSV function and a PropertyReadFile for reading the locations & values. Then use the function in a GenericFunctionMaterial
